cwl_wes/custom_config.py

```diff
@@ -3,22 +3,107 @@
 
+
 class StorageConfig(FOCABaseConfig):
+    """Model for task run and storage configuration.
+
+    Args:
+        tmp_dir: Temporary run directory path
+        permanent_dir: Permanent working directory path
+        remote_storage_url: Remote file storage FTP endpoint
+
+    Attributes:
+        tmp_dir: Temporary run directory path
+        permanent_dir: Permanent working directory path
+        remote_storage_url: Remote file storage FTP endpoint
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> StorageConfig(
+        ...     tmp_dir='/data/tmp',
+        ...     permanent_dir='/data/output',
+        ...     remote_storage_url='ftp://ftp.private/upload'
+        ... )
+        StorageConfig(tmp_dir='/data/tmp', permanent_dir='/data/output', remote_storage_url='ftp://ftp.private/upload')
+    """
     permanent_dir: str = '/data/output'
     tmp_dir: str = '/data/tmp'
     remote_storage_url: str = 'ftp://ftp-private.ebi.ac.uk/upload/foivos'
 
 
 class MonitorConfig(FOCABaseConfig):
+    """Model for monitoring celery.
+
+    Args:
+        timeout: Celery task timeout.
+
+    Attributes:
+        timeout: Celery task timeout.
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> MonitorConfig(
+        ...     timeout=15,
+        ... )
+        MonitorConfig(timeout=15)
+    """
     timeout: float = 0.1
 
 
 class CeleryConfig(FOCABaseConfig):
+    """Model for celery configurations.
+
+    Args:
+        monitor: Celery monitoring configurations.
+        message_maxsize: Celery message max size.
+
+    Attributes:
+        monitor: Celery monitoring configurations.
+        message_maxsize: Celery message max size.
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> CeleryConfig(
+        ...     monitor=MonitorConfig(timeout=15),
+        ...     message_maxsize=1024
+        ... )
+        CeleryConfig(monitor=MonitorConfig(timeout=15), message_maxsize=1024)
+    """
     monitor: MonitorConfig = MonitorConfig()
     message_maxsize: int = 16777216
 
 
 class EndpointConfig(FOCABaseConfig):
+    """Model for celery configurations.
+
+    Args:
+        monitor: Celery monitoring configurations.
+        message_maxsize: Celery message max size.
+
+    Attributes:
+        monitor: Celery monitoring configurations.
+        message_maxsize: Celery message max size.
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> CeleryConfig(
+        ...     monitor=MonitorConfig(timeout=15),
+        ...     message_maxsize=1024
+        ... )
+        CeleryConfig(monitor=MonitorConfig(timeout=15), message_maxsize=1024)
+    """
```

Comment:
Copy/paste issue: this docstring refers to the class above this one. Please replace with the one for the endpoints